library/manageiq_user_management.py

```diff
@@ -0,0 +1,259 @@
+            resource = {'userid': userid, 'name': username, 'password': password,
+                        'group': {'id': group_id}}
+            result = self.client.post(url, action='edit', resource=resource)
+            self.changed = True
+            return result
+        except Exception as e:
+            self.module.fail_json(msg="Failed to update user {userid}: {error}".format(userid=userid, error=e))
+
+    def create_user(self, userid, username, group_id, password):
+        """Creates the user in manageiq.
+
+        Returns:
+            the created user id, name, created_on timestamp,
+            updated_on timestamp, userid and current_group_id
+        """
+        try:
+            url = '{api_url}/users'.format(api_url=self.api_url)
+            resource = {'userid': userid, 'name': username, 'password': password,
+                        'group': {'id': group_id}}
+            result = self.client.post(url, action='create', resource=resource)
+            self.changed = True
+            return result['results']
+        except Exception as e:
+            self.module.fail_json(msg="Failed to create user {userid}: {error}".format(userid=userid, error=e))
+
+    def create_or_update_user(self, userid, username, password, group):
+        """ Create or update a user in manageiq.
+
+        Returns:
+            Whether or not a change took place and a message describing the
+            operation executed.
+        """
+        group_id = self.find_group_by_name(group)
+        if not group_id:  # group doesn't exist
+            self.module.fail_json(
+                msg="Failed to create user {userid}: group {group_name} does not exist in manageiq".format(userid=userid, group_name=group))
+
+        user_id = self.find_user_by_userid(userid)
+        if user_id:  # user already exist
+            if not self.user_update_required(user_id, userid, username, group_id):
+                return dict(
+                    changed=self.changed,
+                    msg="User {userid} already exist, no need for updates".format(userid=userid))
+            else:  # updates required
+                results = self.update_user(user_id, userid, username, group_id, password)
+                return dict(
+                    changed=self.changed,
+                    msg="Successfully updated the user {userid}: {user_details}".format(userid=userid, user_details=results))
+
+        results = self.create_user(userid, username,  group_id, password)
+
+        return dict(
+            changed=self.changed,
+            msg="Successfully created the user {userid}: {user_details}".format(userid=userid, user_details=results))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            userid=dict(required=True, type='str'),
+            username=dict(required=False, type='str'),
+            password=dict(required=False, type='str', no_log=True),
+            group=dict(required=False, type='str'),
```

Comment:
As we discussed, multiple groups is probably going to be what we want.  Change to a list.